### PR TITLE
InstanceDataframe for maps (strictly additive, does not remove InstanceTable)

### PR DESCRIPTION
**Pull request by AhmetNSimsek**
This PR adds an instance table in the form of a data frame for maps (#329). It is strictly additive and does not remove any existing functionality for the other instance tables of class `InstanceTable`.
While more discussion and additions could be had, this is a highly requested feature. Therefore, IMO we should add it as is and continue discussing the remainder under a new issue.
(`InstanceDataframe` could be expanded with similar functionalities of `InstanceTable`, however, for the time bein the goal is to provide an overview. Also, `query` provides a lot of functionalites.)
Usage:
```python
import siibra
print(siibra.maps.head())
print(siibra.maps.columns)
print(siibra.maps.index)
siibra.maps.query("space == 'MNI Colin 27'")
```

### CODE REVIEW

**Comment by AhmetNSimsek** [comment]
Note: iterator of dataframes returns headers hence loops utilizing the map registry do not work. A simple iterator of the form
```python
def __iter__(self) -> Iterator[T]:
        yield from self._objs.values()
```
enables this. However, then it breaks the default representation of `pandas.Dataframes`.

File: siibra/commons.py [hunk 1]
```diff
@@ -215,6 +217,53 @@ def __getattr__(self, index) -> T:
...
+            # allow to build a filtered instance table
+            df = pd.DataFrame.query(self, expr, inplace=False)
+            return self.__class__(
+                objs={
+                    k: v for k, v in self._objs.items()
+                    if k in df.index
+                },
+                attrs=list(dict(df.T).values())
+            )
+
```
Comment (xgui3783) [inline siibra/commons.py:266]: What's the benefit of having a bespoke class for the dataframe? couldn't `.dataframe` just return a dataframe?

File: siibra/commons.py [hunk 2]
```diff
@@ -215,6 +217,53 @@ def __getattr__(self, index) -> T:
...
+            for i, val in enumerate(values):
+                attrs.append({'name': val.name, 'species': str(val.species)})
+                if hasattr(val, 'maptype'):
+                    attrs[i].update(
+                        {
+                            attribute: val.__getattribute__(attribute).name
+                            for attribute in ['parcellation', 'space', 'maptype']
+                        }
+                    )
+            self._dataframe_cached = self.InstanceDataframe(self._elements, attrs)
```
Comment (xgui3783) [inline siibra/commons.py:234]: (cont'ed)

so we would have:

```python
self._dataframe_cached = pd.DataFrame(index=self._elements.keys(), data=attrs)
```

here

File: siibra/commons.py (hunk 1 above)
Comment (AhmetNSimsek) [reply siibra/commons.py:266]: I wanted to preserve some extra functionality like autocomplete but in hind sight, I see that one can just use the standard registry for such functionality.

**Review by xgui3783 — APPROVED** [review]
LGTM

**Pull request merged**